### PR TITLE
Cloud formation removal of unsupported and supported resources 

### LINKED ISSUE #6009: bug: update CDK exists stack resources on next run
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

I have my code here, its working perfect first time i run cdklocal deploy, if i run it again and remove eg. s3 bucket or my codeartifact domain its will sending a lots of error in my docker container.

```python
from constructs import Construct
from aws_cdk import (
    App, Stack,
    aws_codeartifact as codeartifact,
    aws_s3 as s3,
)


class LocalTestStack(Stack):
    def __init__(self, scope: Construct, id: str, **kwargs):
        super().__init__(scope, id, **kwargs)

        cfn_domain = codeartifact.CfnDomain(
            self,
            "TestCodeArtifactDomain",
            domain_name="DemoDomain"
        )

        s3_bucket_test = s3.Bucket(self, 'TestBucket')
        s3_bucket_test.grant_public_access()

        s3_bucket_test2 = s3.Bucket(self, 'HestBucket')
        s3_bucket_test2.grant_public_access()


app = App()
LocalTestStack(app, "LocalTestStack")
app.synth()
```

my log look like this, look like some localhost proxy post request there are failed.

```
2022-05-04T12:56:44.202:WARNING:localstack.utils.server.http2_server: Error in proxy handler for request POST http://localhost:4566/: 'ParameterValue' Traceback (most recent call last):

  File "/opt/code/localstack/localstack/utils/server/http2_server.py", line 185, in index

    result = await run_sync(handler, request, data)

  File "/opt/code/localstack/localstack/utils/asyncio.py", line 85, in run_sync

    return await loop.run_in_executor(thread_pool, copy_context().run, func_wrapped)

  File "/opt/code/localstack/localstack/utils/threads.py", line 39, in run

    result = self.func(self.params, **kwargs)

  File "/opt/code/localstack/localstack/utils/asyncio.py", line 30, in _run

    return fn(*args, **kwargs)

  File "/opt/code/localstack/localstack/services/generic_proxy.py", line 973, in handler

    return modify_and_forward(

  File "/opt/code/localstack/localstack/services/generic_proxy.py", line 514, in wrapper

    value = func(*args, **kwargs)

  File "/opt/code/localstack/localstack/services/generic_proxy.py", line 594, in modify_and_forward

    listener_result = listener.forward_request(

  File "/opt/code/localstack/.venv/lib/python3.10/site-packages/localstack_ext/services/edge.py.enc", line 132, in forward_request

  File "/opt/code/localstack/localstack/services/edge.py", line 170, in forward_request

    result = do_forward_request(api, method, path, data, headers, port=port)

  File "/opt/code/localstack/localstack/utils/patch.py", line 38, in proxy

    return new(target, *args, **kwargs)

  File "/opt/code/localstack/.venv/lib/python3.10/site-packages/localstack_ext/services/edge.py.enc", line 128, in do_forward_request

  File "/opt/code/localstack/localstack/services/edge.py", line 225, in do_forward_request

    result = do_forward_request_inmem(api, method, path, data, headers, port=port)

  File "/opt/code/localstack/localstack/services/edge.py", line 249, in do_forward_request_inmem

    response = modify_and_forward(

  File "/opt/code/localstack/localstack/services/generic_proxy.py", line 514, in wrapper

    value = func(*args, **kwargs)

  File "/opt/code/localstack/localstack/services/generic_proxy.py", line 594, in modify_and_forward

    listener_result = listener.forward_request(

  File "/opt/code/localstack/localstack/http/adapters.py", line 51, in forward_request

    response = self.request(request)

  File "/opt/code/localstack/localstack/aws/proxy.py", line 42, in request

    response = self.skeleton.invoke(context)

  File "/opt/code/localstack/localstack/aws/skeleton.py", line 153, in invoke

    return self.dispatch_request(context, instance)

  File "/opt/code/localstack/localstack/aws/skeleton.py", line 165, in dispatch_request

    result = handler(context, instance) or {}

  File "/opt/code/localstack/localstack/aws/skeleton.py", line 117, in __call__

    return self.fn(*args, **kwargs)

  File "/opt/code/localstack/localstack/aws/api/core.py", line 95, in operation_marker

    return fn(*args, **kwargs)

  File "/opt/code/localstack/localstack/services/cloudformation/provider.py", line 951, in execute_change_set

    deployer.apply_change_set(change_set)

  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1219, in apply_change_set

    self.apply_changes(

  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1490, in apply_changes

    self.init_resource_status(old_resources, action="UPDATE")

  File "/opt/code/localstack/localstack/utils/cloudformation/template_deployer.py", line 1338, in init_resource_status

    stack.set_resource_status(resource_id, "%s_IN_PROGRESS" % action)

  File "/opt/code/localstack/localstack/services/cloudformation/provider.py", line 183, in set_resource_status

    self._set_resource_status_details(resource_id, physical_res_id=physical_res_id)

  File "/opt/code/localstack/localstack/services/cloudformation/provider.py", line 192, in _set_resource_status_details

    resource = self.resources.get(resource_id)

  File "/opt/code/localstack/localstack/services/cloudformation/provider.py", line 255, in resources

    add_params(defaults=False)

  File "/opt/code/localstack/localstack/services/cloudformation/provider.py", line 250, in add_params

    else param["ParameterValue"]

KeyError: 'ParameterValue'
```

### Expected Behavior

Expect to keep my stack up to date, its very important when you need to test a new configuration from time to time, and not a "fresh" run.

### How are you starting LocalStack?

With the `localstack` script

### Steps To Reproduce

Steps are wirte in the start.

### Environment

```markdown
- OS: macOS 12.3.1
- LocalStack: 0.14.2.7
- cdklocal: 2.22.0 (build 1db4b16)
- Docker Desktop: 4.7.1
- Docker Engine: 20.10.14
```

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by lukqw**
This PR adds a test which replicates the scenario of issue #6009.

Essentially, deploying a stack with an unsupported resource (in this case CodeArtifact) is possible and yields no errors.
However, trying to deploy an Update ChangeSet, which removes the previously "created" CodeArtifact fails, as it is not supported.

Users probably either expect a scenario in which both deployments fail or succeed.

This PR accomplishes this change, and only tries to remove resources that are supported.

Closes & fixes #6009

### CODE REVIEW

File: tests/integration/cloudformation/test_cloudformation_changesets.py [hunk 1]
```diff
@@ -373,3 +373,21 @@ def test_delete_change_set_nonexisting(cfn_client):
...
     assert ex.value.response["Error"]["Code"] == "ResourceNotFoundException"
+
+
+@pytest.mark.xfail
+@pytest.mark.aws_validated
+def test_deploy_create_and_then_remove_non_supported_resource_change_set(deploy_cfn_template):
+    # first deploy cfn with a CodeArtifact resource that is not actually supported
+    stack = deploy_cfn_template(
+        template=load_template_raw("code_artifact_template.yaml"),
+        parameters={"IncludeCodeArtifact": "true", "CADomainName": f"domainname-{short_uid()}"},
```
[the remark this answers is not in the document]
Comment (dominikschubert) [reply tests/integration/cloudformation/test_cloudformation_changesets.py:384]: The conditions actually work (to some extent) so I'd leave the single template version. One thing I can imagine tripping the engine up is the use of booleans but with the explicit `Type: String`.

[the remark this answers is not in the document]
Comment (lukqw) [reply tests/integration/cloudformation/test_cloudformation_changesets.py]: You were right. I tried it with something other than true|false and it works now...

**Problem:**
The exception is actually not thrown when sticking to the single template version. 
You can take a look at below logs and compare them if you have the time.

**Pasting logs from the two templates version:**

deploy step (first use of deploy_cfn_template)
```
WARNING  localstack.utils.cloudformation.template_deployer:template_deployer.py:828 Unexpected resource type CodeArtifact::Domain when resolving references of resource TestCodeArtifactDomain: {"Type": "AWS::CodeArtifact::Domain", "Metadata": {"aws:cdk:path": "UpdateCdkStack/TestCodeArtifactDomain"}, "LogicalResourceId": "TestCodeArtifactDomain", "Properties": {"DomainName": "domainname-add0a602"}}. To find out if CodeArtifact::Domain is supported in LocalStack Pro, please check out our docs at https://docs.localstack.cloud/aws/cloudformation
WARNING  localstack.utils.cloudformation.template_deployer:template_deployer.py:1260 Unable to deploy resource type "CodeArtifact::Domain": {"Type": "AWS::CodeArtifact::Domain", "Metadata": {"aws:cdk:path": "UpdateCdkStack/TestCodeArtifactDomain"}, "LogicalResourceId": "TestCodeArtifactDomain", "Properties": {"DomainName": "domainname-add0a602"}}
DEBUG    localstack.utils.cloudformation.template_deployer:template_deployer.py:1598 Handling "Add" for resource "TestCodeArtifactDomain" (1/6) type "AWS::CodeArtifact::Domain" in loop iteration 1 (should_deploy=False)
```

remove step (second use of deploy_cfn_template)
```
2022-05-31T21:43:33.637:WARNING:localstack.utils.cloudformation.template_deployer: Action delete for resource type CodeArtifact::Domain not available. To find out if CodeArtifact::Domain is supported in LocalStack Pro, please check out our docs at https://docs.localstack.cloud/aws/cloudformation
2022-05-31T21:43:33.639:DEBUG:localstack.utils.cloudformation.template_deployer: Extract resource attribute: CodeArtifact::Domain PhysicalResourceId
2022-05-31T21:43:33.641:WARNING:localstack.utils.cloudformation.template_deployer: Unexpected resource type CodeArtifact::Domain when resolving references of resource TestCodeArtifactDomain: {"Type": "AWS::CodeArtifact::Domain", "Metadata": {"aws:cdk:path": "UpdateCdkStack/TestCodeArtifactDomain"}, "LogicalResourceId": "TestCodeArtifactDomain", "Properties": {"DomainName": "domainname-add0a602"}, "PhysicalResourceId": null}. To find out if CodeArtifact::Domain is supported in LocalStack Pro, please check out our docs at https://docs.localstack.cloud/aws/cloudformation
2022-05-31T21:43:33.648:DEBUG:localstack.utils.cloudformation.template_deployer: Dependencies for "TestCodeArtifactDomain" not yet satisfied, retrying in next loop: Unable to fetch details for resource "TestCodeArtifactDomain" (attribute "PhysicalResourceId")
```

later leads to the Exception
```
2022-05-31T21:43:41.967:DEBUG:localstack.utils.cloudformation.template_deployer: Error applying changes for CloudFormation stack "stack-b5a8bfa9": Resource deployment loop completed, pending resource changes: [{'Type': 'Resource', 'ResourceChange': {'Action': 'Remove', 'LogicalResourceId': 'TestCodeArtifactDomain', 'PhysicalResourceId': None, 'ResourceType': 'AWS::CodeArtifact::Domain', 'Replacement': 'False', 'ChangeSetId': None}}] Traceback (most recent call last):
  File "/home/lu/localstack/localstack/localstack/utils/cloudformation/template_deployer.py", line 1546, in _run
    self.do_apply_changes_in_loop(changes, stack, stack_name)
  File "/home/lu/localstack/localstack/localstack/utils/cloudformation/template_deployer.py", line 1630, in do_apply_changes_in_loop
    raise Exception(
Exception: Resource deployment loop completed, pending resource changes: [{'Type': 'Resource', 'ResourceChange': {'Action': 'Remove', 'LogicalResourceId': 'TestCodeArtifactDomain', 'PhysicalResourceId': None, 'ResourceType': 'AWS::CodeArtifact::Domain', 'Replacement': 'False', 'ChangeSetId': None}}]
```

**Logs from the single template version instead:**

deploy step (first use of deploy_cfn_template)
```
2022-05-31T21:56:23.797:WARNING:localstack.utils.cloudformation.template_deployer: Unexpected resource type CodeArtifact::Domain when resolving references of resource TestCodeArtifactDomain: {"Type": "AWS::CodeArtifact::Domain", "Metadata": {"aws:cdk:path": "UpdateCdkStack/TestCodeArtifactDomain"}, "Condition": "DeployCodeArtifact", "LogicalResourceId": "TestCodeArtifactDomain", "Properties": {"DomainName": "domainname-d46850ab"}}. To find out if CodeArtifact::Domain is supported in LocalStack Pro, please check out our docs at https://docs.localstack.cloud/aws/cloudformation
2022-05-31T21:56:23.799:WARNING:localstack.utils.cloudformation.template_deployer: Unable to deploy resource type "CodeArtifact::Domain": {"Type": "AWS::CodeArtifact::Domain", "Metadata": {"aws:cdk:path": "UpdateCdkStack/TestCodeArtifactDomain"}, "Condition": "DeployCodeArtifact", "LogicalResourceId": "TestCodeArtifactDomain", "Properties": {"DomainName": "domainname-d46850ab"}}
2022-05-31T21:56:23.800:DEBUG:localstack.utils.cloudformation.template_deployer: Handling "Add" for resource "TestCodeArtifactDomain" (1/6) type "AWS::CodeArtifact::Domain" in loop iteration 1 (should_deploy=False)
``` 

remove step (second use of deploy_cfn_template)
```
2022-05-31T21:57:11.435:DEBUG:localstack.utils.cloudformation.template_deployer: Skipping deployment of "TestCodeArtifactDomain", as resource condition evaluates to false
2022-05-31T21:57:11.437:DEBUG:localstack.utils.cloudformation.template_deployer: Handling "Modify" for resource "TestCodeArtifactDomain" (1/6) type "AWS::CodeArtifact::Domain" in loop iteration 1 (should_deploy=None)
```

Seems like the first deployment is equivalent, but "skipping" the resource in the single template version does something different than when that resource is not included in the template at all.

[the remark this answers is not in the document]
Comment (dominikschubert) [reply tests/integration/cloudformation/test_cloudformation_changesets.py]: Seems this is still verifying the proper behavior and it's just the CloudFormation engine not handling it correctly. Given the information you provided I'd actually argue that it makes sense to keep both tests "versions" for now. :+1:

**Review by dominikschubert — APPROVED** [review]
LGTM :+1: 

Definitely good to have a "soft" fallback here for now until we have more delete support :+1:

**Pull request merged**